Panel.c

```diff
@@ -503,5 +506,39 @@ int Panel_getCh(Panel* this) {
 #ifdef HAVE_SET_ESCDELAY
    set_escdelay(25);
 #endif
-   return getch();
+
+#ifdef HAVE_LIBNCURSESW
+   static char mbbuf[MB_LEN_MAX];
+   static size_t mbbuf_len = 0, mbbuf_i = 0;
+
+   if (mbbuf_i < mbbuf_len) {
+      return mbbuf[mbbuf_i++];
+   }
+
+   wint_t wkey;
+   int wchrv = get_wch(&wkey);
+   wchar_t wch = (wchar_t)wkey;
+
+   if (wchrv == OK) {
+      mbbuf_i = 0;
+      mbbuf_len = wcrtomb(mbbuf, wch, NULL);
+      if (mbbuf_len == (size_t)-1) {
+         mbbuf_len = 0;
+         return wkey;
+      }
+      last_wkey = wkey;
```

Comment:
I tried to re-review this patch and I have some more comments on this:
1. I think we should upgrade the `Panel_getCh` function to `Panel_getWch` to avoid a state buffer in this function, making things more fragile.
2. Since this is the primary function we read user's keyboard input, if we want to sanitize user input, this would be the best place to do.
3. However, I've changed my mind on filtering all unprintable characters. Instead, filter only (a.) invalid characters, that is, when `wctomb` would fail to convert the character, and (b.) newlines of all kind. Replace the character with U+FFFD if `wctomb` fails. Don't replace other Unicode characters because user might input it through a paste from the clipboard to the terminal emulator, and those characters should be intact as much as possible.